crates/engine/tree/src/tree/mod.rs

```diff
@@ -1175,7 +1175,18 @@ where
             return Ok(valid_outcome(state.head_block_hash))
         }
 
-        // 2. ensure we can apply a new chain update for the head block
+        // 2. Check if forkchoice state points to a valid ancestor of the canonical head.
+        if self.is_valid_canonical_ancestor(state.head_block_hash)? {
+            trace!(
+                target: "engine::tree",
+                head=?state.head_block_hash,
+                "fcu head block is a valid ancestor of the canonical head, skipping update"
+            );
+
+            return Ok(valid_outcome(state.head_block_hash));
+        }
```

Comment:
this check makes sense, but I know wonder if we're not actually doing this already in

> // 4. check if the head is already part of the canonical chain

this actually clashes with with this opstack specialization that we have here which I forgot about:

https://github.com/paradigmxyz/reth/blob/03c5abba887452a00680ab5e8a1aee34811158a1/crates/engine/tree/src/tree/mod.rs#L1202-L1204